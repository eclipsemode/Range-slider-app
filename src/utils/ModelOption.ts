declare type ModelOption = {
    from: number,
    to: number,
    gap: number,
    step: number,
    min: number,
    max: number,
    vertical: boolean;
    rulers: boolean,
    range: boolean,
    tooltip: boolean,
    percent: boolean,
    color: {
        firstColor: string,
        secondColor: string,
        textColor: string,
        thumbColor: string
    },
    configPanel: boolean,
    controlConfig: string[],
    toggleConfig: string[],
    progress: boolean
}

export default ModelOption;