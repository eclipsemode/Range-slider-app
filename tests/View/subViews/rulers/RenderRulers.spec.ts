import $ from "jquery";
import ModelOption from "../../../../src/utils/ModelOption";
import {
  CreateBar,
  CreateRulers,
  RenderRulers,
} from "../../../../src/View/subViews";

describe("Render Rulers tests", () => {
  const body: JQuery = $("body");
  const bar: CreateBar = new CreateBar(body);
  const valuesArr: number[] = [1, 10, 20, 30, 40, 50];
  const pixelsArr: number[] = [1, 10, 20, 30, 40, 50];
  const options: ModelOption = {
    max: 104,
    tooltip: true,
    range: true,
    configPanel: true,
    step: 33,
    gap: 1,
    min: 0,
    from: 0,
    to: 99,
    vertical: false,
    rulers: true,
    percent: false,
    color: {
      firstColor: "#ffe53b",
      secondColor: "#ff2525",
      textColor: "#ffe53b",
      thumbColor: "#ffe53b",
    },
    progress: true,
    controlConfig: ["min", "max", "step", "from", "to"],
    toggleConfig: ["vertical", "range", "rulers", "progress", "tooltip"],
  };
  const rulers: CreateRulers = new CreateRulers(
    body,
    valuesArr,
    pixelsArr,
    false
  );
  test("Should be defined", () => {
    const renderRulers: RenderRulers = new RenderRulers(options, rulers, bar);
    expect(renderRulers.rulers).toBeDefined();
  });
  test("Should match snapshot vertical", () => {
    const renderRulers: RenderRulers = new RenderRulers(
      { ...options, vertical: true },
      rulers,
      bar
    );
    expect(renderRulers.rulers.rulersElement[0].innerHTML).toMatchSnapshot(
      "vertical"
    );
  });
  test("Should be null", () => {
    const renderRulers: RenderRulers = new RenderRulers(
      { ...options, rulers: false },
      rulers,
      bar
    );
    expect(renderRulers.rulers).toBeNull();
  });
  test("Should match values", () => {
    const renderRulers: RenderRulers = new RenderRulers(options, rulers, bar);
    expect(renderRulers.rulers.rulersElement[0].innerHTML).toMatch(
      "slider-app__rulers-value"
    );
  });
});
