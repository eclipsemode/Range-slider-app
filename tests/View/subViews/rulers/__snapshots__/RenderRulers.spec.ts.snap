// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Rulers tests Should match snapshot vertical: vertical 1`] = `"<div class=\\"slider-app__rulers-value slider-app__rulers-value--0 slider-app__rulers-value--vertical\\" style=\\"bottom: 0px;\\">104</div>"`;
